test/addons/async-hooks-id/binding.cc

```diff
@@ -17,9 +17,17 @@ void GetTriggerAsyncId(const FunctionCallbackInfo<Value>& args) {
     node::AsyncHooksGetTriggerAsyncId(args.GetIsolate()));
 }
 
+void EmitAsyncInit(const FunctionCallbackInfo<Value>& args) {
+  assert(args[0]->IsObject());
+  node::async_uid uid =
+      node::EmitAsyncInit(args.GetIsolate(), args[0].As<Object>(), "foobar");
```

Comment:
> The `EmitAsyncInit` that returns `async_id` requires 4 parameters, correct? Or is this a test for the `async_context` casting?

Yes, the latter. The old one is not really exposed through our API anymore, it’s just the symbol that has to keep being exported so already-compiled libraries can reference it.

> Also, is `node::async_uid` by purpose?

Yes, everywhere else in the tests we use `async_id` now, so there should be one place where we don’t.